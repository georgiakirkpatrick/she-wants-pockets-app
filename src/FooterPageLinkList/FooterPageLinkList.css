ul {
    list-style-type: none;
}

.Footer__page-link-list {
    display: flex;
    list-style-type: none;
}