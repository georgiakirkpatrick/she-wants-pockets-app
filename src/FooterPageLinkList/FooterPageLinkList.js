import React from 'react';
import FooterPageLink from '../FooterPageLink/FooterPageLink';
import './FooterPageLinkList.css';

function FooterPageLinkList(props) {
    const generateFooterLinks = props.footerLinks.map(link => (
        <FooterPageLink 
            key={link.id}
            route={link.route}
            name={link.name}
        />
    ))

    return <nav className='Footer__page-link-list'>
        <ul>
            {generateFooterLinks}
        </ul>
    </nav>
}

export default FooterPageLinkList;