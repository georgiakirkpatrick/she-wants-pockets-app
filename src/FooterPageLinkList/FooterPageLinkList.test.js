import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import FooterPageLinkList from './FooterPageLinkList';

const testLinks = [  
    {
        id: 10101,
        pathToProfile: 'https://www.instagram.com/shewantspockets/',
        pathToImage: 'https://images.app.goo.gl/8ophL7f6ksAuhBZR6',
        platform: 'Instagram'
    }
];

describe ('FooterPageLinkList', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<FooterPageLinkList footerLinks={testLinks} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it ('renders the UI as expected', () => {
        const tree = renderer
            .create(<FooterPageLinkList footerLinks={testLinks} />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    });
});    