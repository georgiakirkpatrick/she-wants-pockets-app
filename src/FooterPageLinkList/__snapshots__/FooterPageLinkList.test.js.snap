// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterPageLinkList renders the UI as expected 1`] = `
<nav
  className="Footer__page-link-list"
>
  <ul>
    <li
      className="Footer__page-link"
    >
      <a />
    </li>
  </ul>
</nav>
`;
